src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -410,14 +479,18 @@ module.exports = class LevelPlane {
       if (block.blockType === "doorIron" && position !== positionInQuestion) {
         block.isPowered = this.powerCheck(position, true);
         if (block.isPowered && !block.isOpen) {
-          block.isOpen = true;
           if (this.levelModel) {
-            this.levelModel.controller.levelView.animateDoor(index, true);
+            if (!this.skipDoorFrom(position)) {
```

Comment:
I wonder if it would be simpler to have the door check if any entity is present at the current index.  If yes, add a listener to the entity (or entities) to check again on their next movement.  What do you think?

Reply:
There are multiple ways the door can try to close, beyond a player moving off a pressure plate.  Here's another example: https://github.com/code-dot-org/craft/issues/118.